src/psyclone/tests/psyGen_test.py

```diff
@@ -5060,25 +5060,102 @@ def test_fp2astproc_case_default(f2008_parser):
                           Assignment)
 
 
-def test_fp2astproc_handling_invalid_case_construct(f2008_parser):
-    ''' Test that the Case_Construct handler raises the proper errors when
-    it parses invalid or unsupported fparser2 trees.
-    '''
+def test_fp2astproc_handling_case_list(f2008_parser):
+    ''' Test that the Case_Construct handler correctly processes CASE
+    statements involving a list of conditions. '''
     from fparser.common.readfortran import FortranStringReader
     from fparser.two.Fortran2003 import Execution_Part
+    reader = FortranStringReader(
+        '''SELECT CASE (my_var)
+            CASE (label2, label3)
+                branch2 = 1
+            END SELECT''')
+    fparser2case_construct = Execution_Part.match(reader)[0][0]
 
-    # CASE Value Ranges are not supported
+    fake_parent = Node()
+    processor = Fparser2ASTProcessor()
+    processor.process_nodes(fake_parent, [fparser2case_construct], None)
+    assert len(fake_parent.children) == 1
+    ifnode = fake_parent.children[0]
```

Comment:
I'm not sure if it would be any better but if you want to check the tree you could compare with `str(fake_parent.children[0])` for this test and subsequent case tests.

Reply:
It would be shorter but it would then mean I was testing the Fortran backend as well as the processor.

Reply:
Yes, good point.